File: api/services/resumeParser.js
import fs from 'fs/promises';
import pdfParser from 'pdf-parse';

export async function parseResume(filePath) {
  try {
    const dataBuffer = await fs.readFile(filePath);
    const parsed = await pdfParser(dataBuffer);

    await fs.unlink(filePath);

    return parsed.text;
  } catch (err) {
    console.error("❌ Failed to parse or delete file:", err);
    throw err;
  }
}

File: api_fastapi/services/resumeParser.py
import pdfplumber
import os


def parse_resume(file_path: str) -> str:
    try:
        with pdfplumber.open(file_path) as pdf:
            text = "\n".join(page.extract_text() or "" for page in pdf.pages)

        os.remove(file_path)
        return text.strip()

    except Exception as err:
        print("Failed to parse or delete file:", err)
        raise
